docs/source/_ext/dissect_plugins.py

```diff
@@ -0,0 +1,149 @@
+from pathlib import Path
+from typing import Any
+
+from dissect.target.helpers import docs
+from dissect.target.plugin import load, plugins
+from sphinx.application import Sphinx
+from sphinx.util.console import colorize
+from sphinx.util.display import status_iterator
+from sphinx.util.logging import getLogger
+
+LOGGER = getLogger(__name__)
+
+PAGE_TEMPLATE = """..
+    generated, remove this comment to keep this file
+
+{title}
+
+.. code-block:: console
+
+    $ target-query <path/to/target> -f {name}
+
+{variants}
+"""
+
+VARIANT_TEMPLATE = """
+.. list-table:: Details
+    :widths: 20 80
+
+    * - Module
+      - ``{module}.{class}``
+    * - Output
+      - ``{output}``
+
+**Module documentation**
+
+{class_doc}
+
+**Function documentation**
+
+{func_doc}
+"""
+
+NAMESPACE_TEMPLATE = """
+This is a namespace plugin. This means that by running this plugin, it will automatically run
+all other plugins under this namespace:
+
+{exports}
+"""
+
+
+def builder_inited(app: Sphinx) -> None:
+    dst = Path(app.srcdir).joinpath("plugins")
+    if not dst.exists():
+        dst.mkdir()
+
+    plugin_map = {}
+    for plugin in plugins():
+        # Ignore all modules in general as those are all internal or utility
+        if plugin["module"].startswith("general."):
+            continue
+
+        if plugin["namespace"]:
+            plugin_map.setdefault(plugin["namespace"], []).append(plugin)
+
+        for export in plugin["exports"]:
+            if export == "get_all_records":  # TODO we need to remove this
+                continue
+
+            if plugin["namespace"]:
+                export = f"{plugin['namespace']}.{export}"
+            plugin_map.setdefault(export, []).append(plugin)
+
+    for name, plugin in status_iterator(
+        plugin_map.items(),
+        colorize("bold", "[Dissect] Writing plugin files... "),
+        length=len(plugin_map),
+        stringify_func=(lambda x: x[0]),
+    ):
+        dst_path = dst.joinpath(name + ".rst")
+        if dst_path.exists():
+            continue
+
+        dst_path.write_text(_format_template(name, plugin))
+
+
+def build_finished(app: Sphinx, exception: Exception) -> None:
+    if not app.config.dissect_plugins_keep_files:
+        dst = Path(app.srcdir).joinpath("plugins")
+        if app.verbosity > 1:
+            LOGGER.info(
+                colorize("bold", "[Dissect] ")
+                + colorize("darkgreen", "Cleaning generated .rst files")
+            )
+
+        for rst in dst.glob("*.rst"):
+            with open(rst, "rb") as fh:
+                if fh.read(16) != b"..\n    generated":
+                    continue
+            rst.unlink()
```

Comment:
```suggestion
    if app.config.dissect_plugins_keep_files:
        return
        
    dst = Path(app.srcdir).joinpath("plugins")
    if app.verbosity > 1:
        LOGGER.info(
            colorize("bold", "[Dissect] ")
            + colorize("darkgreen", "Cleaning generated .rst files")
        )

    for rst in dst.glob("*.rst"):
        with open(rst, "rb") as fh:
            if fh.read(16) != b"..\n    generated":
                continue
        rst.unlink()
```

Reply:
This is setup deliberately this way so that you can define different behaviour for different configuration options. This is a common pattern in Sphinx extensions.

Reply:
Ah, okay, my bad! Never mind then